source/common/common/logger.cc

```diff
@@ -254,6 +256,22 @@ void Registry::setLogFormat(const std::string& log_format) {
   }
 }
 
+void Registry::setJsonLogFormat(const Protobuf::Message& log_format_struct) {
+  Protobuf::util::JsonPrintOptions json_options;
+  json_options.preserve_proto_field_names = true;
+  json_options.always_print_primitive_fields = true;
+
+  std::string format_as_json;
+  if (auto status =
+          Protobuf::util::MessageToJsonString(log_format_struct, &format_as_json, json_options);
+      !status.ok()) {
+    throw EnvoyException(
```

Comment:
I'm pretty certain this won't be used in the data-plane code, but just to be safe here, can this function return absl::Status, and the exception will be thrown in the server's code?

Reply:
done